Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,74 @@ namespace TestImpact
             return testSuites;
         }
     } // namespace GTest
+
+    namespace Python
+    {
+        AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> TestEnumerationSuitesFactory(
+            const AZStd::string& testEnumerationData)
+        {
+            const char newLineDelim = '\n';
+            size_t startLine;
+            size_t endLine = 0;
+            int count = 0;
+
+            AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> pairs;
+            AZStd::map<AZStd::string, AZStd::vector<TestEnumerationSuite>> testSuiteMap;
+            AZStd::string previousTestFixture;
+            // Steps through our string line by line. Assumes lines are split by newline characters.
+            while ((startLine = testEnumerationData.find_first_not_of(newLineDelim, endLine)) != AZStd::string::npos)
```

Comment:
question: Just want to sanity check with threw some weird examples at this to ensure the condition is handled gracefully even if we have weird unexpected data

Reply:
@jonawals Do you have any comments on this as you suggested this snippet to me?